InnovativeCars/force-app/main/default/classes/ShowroomHandler1.cls

```diff
@@ -0,0 +1,98 @@
+/**
+* @author Agnieszka Ząbkowicz
+* @date 06.2021
+*
+* @description handler for ShowroomTrigger
+*/
+public class ShowroomHandler1 {
+
+    /**
+    * @author Agnieszka Ząbkowicz
+    * @date 06.2021
+    *
+    * @description method checking if there is only one warehouse per each account 
+    * and adding error when appropriate
+    * @param newShowrooms list of showroom objects to check
+    */
+    public static void checkIfOnlyOneWarehouse(List<Showroom__c> newShowrooms) {
+            List<Showroom__c> warehouseShowrooms = [SELECT Account__c, Name FROM Showroom__c WHERE Warehouse__c = true];
```

Comment:
Pobierasz wszystkie magazyny z orga jak bedzie ich duzo to error pobierz tylko te ktore sa wazne